### ISSUE TITLE
Question in cut_model.py (bs_per_gpu)

### ISSUE THREAD

**Issue by 1211sh** [taesungp/contrastive-unpaired-translation#100]
![image](https://user-images.githubusercontent.com/33894603/132534803-a5be482e-6a4d-48f3-bf46-f43bda38ae06.png)

Hi,
I am wondering how the multi-gpu process works in 'data_dependent_initialize' function in cut_model.py
The thing is : self.real_A = self.real_A[:bs_per_gpu]

I think self.real_A[:bs_per_gpu] is just simply removing a part of self.real_A, not working as spliting data for multi-gpu.

Is it a typo? or am I wrong?  Please explain me how the multi-gpu works with 'bs_per_gpu'. This thing has been eating me for days.

Thank you.

**Comment by maximiliankukla** [comment]
Hi @1211sh,

I struggled with this as well, but for a different reason

I wasn't looking at this part of the code, but at the NCE loss function, which did behave strangely while debugging, as only half of the inputs were used for a batch_size of 2 (and I would only look at that code the first time the debugger runs into these lines,  repeatedly).

The above code snippet is only run once during initialization, and discarding half of the input. Afterwards in actual training the data is correctly split among the GPUs.
